src/qililab/analog/annealing_program.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+from typing import Any, Callable
+
+import numpy as np
+
+from qililab.platform.components import Bus
+from qililab.waveforms import Arbitrary as ArbitraryWave
+
+
+@dataclass
+class AnnealingProgram:
+    """Class for an Annealing Program. The program should have the format
+
+    [
+        {"qubit_0": {"sigma_x" : 0, "sigma_y" : 1, "sigma_z" : 2},
+         "coupler_1_0 : {...},
+        },      # time=0ns
+        {...},  # time=1ns
+    .
+    .
+    .
+    ]
+    """
+
+    platform: Any
+    anneal_program: list[dict[str, dict[str, float]]]
+
+    def transpile(self, transpiler: Callable):
+        """First implementation of a transpiler, pretty basic but good as a first step. Transpiles from ising coefficients to fluxes
+
+        Args:
+            transpiler (Callable): Transpiler to use. The transpiler should take 2 values as arguments (delta, epsilon)
+            and return 2 values (phix, phiz)
+        """
+
+        # iterate over each anneal step and transpile ising to fluxes
+        for anneal_step in self.anneal_program:
+            for circuit_element in anneal_step:
+                phix, phiz = transpiler(
+                    delta=anneal_step[circuit_element]["sigma_x"], epsilon=anneal_step[circuit_element]["sigma_z"]
+                )
+                anneal_step[circuit_element]["phix"] = phix
+                anneal_step[circuit_element]["phiz"] = phiz
+
+    def get_waveforms(self) -> dict[str, tuple[Bus, ArbitraryWave]]:
+        """Returns a dictionary containing (bus, waveform) for each flux control from the transpiled fluxes. `AnnealingProgram.transpile` should be run first. The waveform is an arbitrary waveform obtained from the transpiled fluxes.
+
+        Returns:
+            dict[str,tuple[Bus,ArbitraryWave]]: Dictionary containing (bus, waveform) for each flux control (i.e. phix or phiz).
+        """
+        # parse names from algorithm notation (e.g. qubit_0) to runcard notation (e.g. q0)
+        element_name_map = {"qubit": "q", "coupler": "c"}
```

Comment:
this is enforcing a runcard standard for naming, no?

Reply:
I mean, what happens if the experimentalist that writes the runcard makes a consistent typo naming 'qubi' all along the runcard?

Reply:
yes true. I could also load the mappings from the runcard so that the standard is set at the runcard (`flux_to_bus_topology`) but some consistency from the user side will still be required when they send the anneal program dictionary however.

Reply:
that is a good point, worth explaining it in the changeloc and documentation, that the annealing schedule mapping needs to be consistent with runcard keys nomenclature.

Reply:
let me clarify this so we can determine the best course of action. The way it's written there are 3 different strings:
- `qubit_0` which contains some attribute / dictionary with keys `phix`. This is used at the higher level because the structure of the entry point (from H algorithm) is of the kind `qubit_0` with some key `Delta` for the ising coefficients. So the transpiler just adds keys `phix` `phiz` under `qubit_0` to the pre-existing dictionary
- `phix_q0` single string used to point to a unique control line. Here we don't do the nested dict structure above because we want to allow flexibility in the runcard (i.e. in the real chip there might not be a "qubit" with unique lines per se, so we want to avoid grouping things under elements like qubit
- `flux_phix_q0` the name of the bus pointing to the `phix_q0` control.
The 1st and 2nd notations are enforced i.e. not following them will make the code break. The 3rd is completely arbitrary, the bus name can be whatever as in the gate based case (as long as there exists a bus with the same alias under `platform.bus`.

Reply:
Cool, let's include this clarifications in the changelog please, so ideally an user that digs into it gets a clear picture.